src/R/Support/Impl/Help/Packages/PackageInfo.cs

```diff
@@ -2,257 +2,113 @@
-                string actualName = mappedNames[mappedName];
-                if (functionIndex.TryGetValue(actualName, out ni)) {
-                    INamedItemInfo niAlias = new NamedItemInfo() {
-                        Name = mappedName,
-                        Description = ni.Description,
-                        ItemType = ni.ItemType
-                    };
-                    functions.Add(niAlias);
-                }
+        private async Task<IEnumerable<string>> GetFunctionNamesAsync() {
+            var cached = TryRestoreFromCache();
+            if (cached == null) {
+                try {
+                    var r = await _host.Session.EvaluateAsync<JArray>(Invariant($"as.list(base::getNamespaceExports('{this.Name}'))"), REvaluationKind.Normal);
+                    return r.Select(p => (string)((JValue)p).Value).ToArray();
+                } catch (MessageTransportException) { } catch (TaskCanceledException) { } catch (REvaluationException) { }
+            } else {
+                _saved = true;
             }
-
-            return functions;
+            return Enumerable.Empty<string>();
         }
 
-        private IReadOnlyDictionary<string, string> GetMappedNames() {
-            Dictionary<string, string> dict = new Dictionary<string, string>();
-
+        /// <summary>
+        /// Attempts to locate cached function list for the package
+        /// </summary>
+        /// <returns></returns>
+        private IEnumerable<string> TryRestoreFromCache() {
+            var filePath = GetCacheFilePath();
             try {
-                string indexFile = Path.Combine(this.InstallPath, this.Name, "help", "AnIndex");
-                if (File.Exists(indexFile)) {
-                    using (StreamReader sr = new StreamReader(indexFile, Encoding.UTF8)) {
-                        string line;
-                        char[] splitChars = new char[] { ' ', '\t' };
-                        while ((line = sr.ReadLine()) != null) {
-                            string[] parts = line.Split(splitChars, StringSplitOptions.RemoveEmptyEntries);
-                            if (parts.Length == 2) {
-                                string mappedName = parts[0];
-                                string actualName = parts[1];
-                                if (mappedName != actualName) {
-                                    dict[mappedName] = actualName;
-                                }
-                            }
+                if (File.Exists(filePath)) {
+                    var list = new List<string>();
+                    using (var sr = new StreamReader(filePath)) {
+                        while (!sr.EndOfStream) {
+                            list.Add(sr.ReadLine().Trim());
                         }
                     }
+                    return list;
                 }
-            } catch (IOException) { }
+            } catch (IOException) { } catch (AccessViolationException) { }
```

Comment:
Did you mean `UnauthorizedAccessException`? `AccessViolationException` is for broken unmanaged pointers.